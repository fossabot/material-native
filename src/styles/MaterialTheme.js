'use strict';
import * as colors from './colors';
import shades from './shades';
const F = Object.freeze;

function normalizeTheme(theme) {
	return theme === 'dark' ? 'dark' : 'light';
}

function inverse(theme) {
	return theme === 'dark' ? 'light' : 'dark';
}

const themeTones = F({
	light: F({
		statusBar: colors.grey300,
		barStyle: 'dark-content',
		appBar: colors.grey100,
		background: colors.grey50,
		container: colors.white,
	}),
	dark: F({
		statusBar: colors.black,
		barStyle: 'light-content',
		appBar: colors.grey900,
		background: '#303030',
		container: colors.grey800,
	}),
});

export default class MaterialTheme {
	static defaultTheme = new MaterialTheme();

	constructor(settings) {
		this.settings = settings || {};
		// @todo recursively freeze settings
		this._applyDefaults();
		this._buildTheme();
		this._freeze();
	}

	_applyDefaults() {
		this.settings.theme = normalizeTheme(this.settings.theme);
	}

	_buildTheme() {
		const {theme} = this.settings;
		const inverseTheme = inverse(theme);

		this.themeTone = theme;
		this.inverseThemeTone = inverseTheme;

		this.palette = {
			primary: this.settings.primary || colors.indigo500,
			darkPrimary: this.settings.darkPrimary || colors.indigo700,
			accent: this.settings.accent || colors.pinkA200,
			background: themeTones[theme].background,
			container: themeTones[theme].container,
		};

		this.defaultToolbar = Object.assign({
			theme,
			background: themeTones[theme].appBar,
		}, this.settings.defaultToolbar || {});

		// @todo Automatically pick the right appBar theme based on the appBar background
		//       Many solid appBar colors should implicitly turn the theme within the appBar to dark
		const appBarTheme = normalizeTheme((this.settings.appBar && this.settings.appBar.theme) || this.defaultToolbar.theme);
		this.appBar = Object.assign({}, this.defaultToolbar, {
			theme: appBarTheme,
			background: this.palette.primary || this.defaultToolbar.background,
		}, this.settings.appBar || {});

		this.statusBar = Object.assign({
			background: this.palette.darkPrimary || themeTones[inverse(appBarTheme)].statusBar,
			translucentBackground: shades.dark.statusBar,
			barStyle: themeTones[inverse(appBarTheme)].barStyle,
		}, this.settings.statusBar || {});

		this.text = Object.assign({
			primaryColor: shades[inverseTheme].primaryText,
			secondaryColor: shades[inverseTheme].secondaryText,
			disabledColor: shades[inverseTheme].disabledText,
			hintColor: shades[inverseTheme].hintText,
			primaryOpacity: shades.opacity[inverseTheme].primaryText,
			secondaryOpacity: shades.opacity[inverseTheme].secondaryText,
			disabledOpacity: shades.opacity[inverseTheme].disabledText,
			hintOpacity: shades.opacity[inverseTheme].hintText,
		}, this.settings.text || {});

		this.icon = Object.assign({
			activeColor: shades[inverseTheme].activeIcon,
			inactiveColor: shades[inverseTheme].inactiveIcon,
			activeOpacity: shades.opacity[inverseTheme].activeIcon,
			inactiveOpacity: shades.opacity[inverseTheme].inactiveIcon,
		}, this.settings.icon || {});
	}

	_freeze() {
		Object.freeze(this); // @todo recursive
	}

	/**
	 * Create a new theme object that inherits settings from another
	 */
	extend(settings) {
		// @todo immutably merge `this.settings` with `settings` and pass that to `new MaterialTheme()`
	}
}

export function getMaterialTheme(settings) {
	return new MaterialTheme(settings);
}
