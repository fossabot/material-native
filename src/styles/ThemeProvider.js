'use strict';
import Observable from 'zen-observable';
import PropTypes from '../PropTypes';
import React, {PureComponent} from 'react';
import MaterialTheme from './MaterialTheme';

const defaultTheme = new MaterialTheme();

export default class ThemeProvider extends PureComponent {
	static contextTypes = {
		materialThemeObservable: PropTypes.Observable,
	};

	static childContextTypes = {
		materialThemeObservable: PropTypes.Observable.isRequired,
		memoizedMaterialStyles: PropTypes.func,
	};

	getChildContext() {
		return {
			materialThemeObservable: this.observable,
			memoizedMaterialStyles: this.memoizedMaterialStyles,
		};
	}

	componentWillMount() {
		// materialTheme
		if ( this.context.materialThemeObservable ) {
			this._parentSubscription = this.context.materialThemeObservable.subscribe({
				next: (theme) => {
					this.parentMaterialTheme = theme;
					this.handleThemeChange(this.props);
				},
				complete: () => {
					delete this._parentSubscription;
				},
			});
		} else {
			this.handleThemeChange(this.props);
		}
	}

	componentWillReceiveProps(nextProps) {
		// materialTheme
		this.handleThemeChange(nextProps);
	}

	componentWillUnmount() {
		if ( this._parentSubscription ) {
			this._parentSubscription.unsubscribe();
		}
	}

	// materialTheme
	observable = new Observable((observer) => {
		if ( this.materialTheme ) {
			observer.next(this.materialTheme);
		}

		this.observers.add(observer);

		return () => {
			this.observers.delete(observer);
		};
	});

	observers = new Set();

	materialTheme = undefined;
	parentMaterialTheme = undefined;

	// materialTheme
	handleThemeChange(props) {
		const baseTheme = props.theme || this.parentMaterialTheme || defaultTheme;

		// @todo Handle extend prop here

		if ( baseTheme !== this.materialTheme ) {
			this.materialTheme = baseTheme;
			this.observers.forEach((observer) => observer.next(this.materialTheme));
		}
	}

	// styleCache
	materialStyleCache = new Map();

	memoizedMaterialStyles = (materialTheme, mapThemeToStyles) => {
		let cache = this.materialStyleCache.get(mapThemeToStyles);
		if ( cache && cache.materialTheme === materialTheme ) {
			return cache.styles;
		}

		// @note If possible in the future, styles in `cache` can be unloaded at this point

		const cacheStyles = mapThemeToStyles(materialTheme);

		for ( const key in cacheStyles ) {
			Object.freeze(cacheStyles[key]);
		}
		Object.freeze(cacheStyles);

		cache = {
			materialTheme,
			styles: cacheStyles,
		};

		this.materialStyleCache.set(mapThemeToStyles, cache);

		return cache.styles;
	};

	// boilerplate
	render() {
		return React.Children.only(this.props.children);
	}
}
