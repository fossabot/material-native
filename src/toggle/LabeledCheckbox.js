'use strict';
import React, {PureComponent} from 'react';
import {StyleSheet, Text} from 'react-native';
import withMaterialTheme from '../styles/withMaterialTheme';
import RectRipple from '../touchable/RectRipple';
import CoreCheckbox from './CoreCheckbox';
import * as typo from '../styles/typo';

/**
 * Checkbox with a label right beside it, useful for things like "remember me" checkboxes
 */
class LabeledCheckbox extends PureComponent {
	static defaultProps = {
		checked: false,
	};

	onPress = (e) => {
		this.props.onPress && this.props.onPress(e);
		this.props.onChangeChecked && this.props.onChangeChecked(!this.props.checked);
	};

	render() {
		const {
			onPress, // eslint-disable-line no-unused-vars
			style,
			label,
			...checkboxProps, // eslint-disable-line comma-dangle
		} = this.props;
		const {
			materialTheme,
		} = checkboxProps;

		const isRTL = false; // @todo support RTL

		const normalProps = materialTheme.checkbox.theme === 'dark' ? {light: true} : {dark: true};

		// @todo Use disabled to disable ripple/press
		return (
			<RectRipple
				{...normalProps}
				pointerEvents='box-only'
				style={[
					style,
					styles.root
				]}
				onPress={this.onPress}>
				<CoreCheckbox
					style={styles.checkbox}
					{...checkboxProps} />
				<Text
					suppressHighlighting
					numberOfLines={1}
					style={[
						styles.label,
						isRTL ? styles.labelRTL : styles.labelLTR,
						{
							color: materialTheme.text.primaryColor,
						}
					]}>
					{label}
				</Text>
			</RectRipple>
		);
	}
}

export default withMaterialTheme(LabeledCheckbox);

const styles = StyleSheet.create({
	root: {
		height: 48,
		flexDirection: 'row',
		alignItems: 'center',
	},
	checkbox: {
		marginHorizontal: (48 - 24) / 2,
	},
	label: {
		...typo.body1,
		marginHorizontal: (48 - 24) / 2,
	},
	labelLTR: {
		marginLeft: 0,
	},
	labelRTL: {
		marginRight: 0,
	},
});
