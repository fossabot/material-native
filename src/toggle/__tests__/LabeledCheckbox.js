'use strict';
import React from 'react';
import MaterialTheme from '../../styles/MaterialTheme';
import ThemeProvider from '../../styles/ThemeProvider';
import LabeledCheckbox from '../LabeledCheckbox';

import ReactTestRenderer from 'react-test-renderer';
import ToJSX from 'react-element-to-jsx-string';

const theme = new MaterialTheme({});

/* eslint-disable react/jsx-key */
const snapshotTests = [
	<LabeledCheckbox />,
	<LabeledCheckbox checked />,
	<LabeledCheckbox indeterminate />,
	<LabeledCheckbox label='Label' />,
	<LabeledCheckbox label='Label' labelColor='#123456' />,
	<LabeledCheckbox label='Label' style={{width: 200}} /> // eslint-disable-line react-native/no-inline-styles
];
/* eslint-enable react/jsx-key */

for ( const root of snapshotTests ) {
	test(ToJSX(root, {maxInlineAttributesLineLength: Infinity}), () => {
		const host = ReactTestRenderer.create(
			<ThemeProvider theme={theme}>
				{root}
			</ThemeProvider>
		);
		expect(host.toJSON()).toMatchSnapshot();
	});
}
