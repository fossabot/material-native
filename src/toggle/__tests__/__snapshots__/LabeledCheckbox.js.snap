// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withMaterialTheme(LabeledCheckbox)
  label="Label"
  style={{
    width: 200
  }}
 /> 1`] = `
<View
  accessible={true}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  rippleLocation="tapLocation"
  style={
    Array [
      Array [
        Object {
          "width": 200,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
        },
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 12,
        },
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          Array [
            undefined,
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginHorizontal": 12,
        },
        Object {
          "marginLeft": 0,
        },
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
      ]
    }
    suppressHighlighting={true}
  >
    Label
  </Text>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "overflow": "hidden",
          "position": "absolute",
        },
        Object {
          "borderRadius": 0,
          "height": 1,
          "width": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<withMaterialTheme(LabeledCheckbox) /> 1`] = `
<View
  accessible={true}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  rippleLocation="tapLocation"
  style={
    Array [
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
        },
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 12,
        },
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          Array [
            undefined,
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginHorizontal": 12,
        },
        Object {
          "marginLeft": 0,
        },
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
      ]
    }
    suppressHighlighting={true}
  />
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "overflow": "hidden",
          "position": "absolute",
        },
        Object {
          "borderRadius": 0,
          "height": 1,
          "width": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<withMaterialTheme(LabeledCheckbox) checked /> 1`] = `
<View
  accessible={true}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  rippleLocation="tapLocation"
  style={
    Array [
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
        },
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 12,
        },
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#3F51B5",
            "fontSize": 24,
          },
          Array [
            undefined,
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginHorizontal": 12,
        },
        Object {
          "marginLeft": 0,
        },
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
      ]
    }
    suppressHighlighting={true}
  />
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "overflow": "hidden",
          "position": "absolute",
        },
        Object {
          "borderRadius": 0,
          "height": 1,
          "width": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<withMaterialTheme(LabeledCheckbox) indeterminate /> 1`] = `
<View
  accessible={true}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  rippleLocation="tapLocation"
  style={
    Array [
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
        },
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 12,
        },
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#3F51B5",
            "fontSize": 24,
          },
          Array [
            undefined,
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginHorizontal": 12,
        },
        Object {
          "marginLeft": 0,
        },
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
      ]
    }
    suppressHighlighting={true}
  />
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "overflow": "hidden",
          "position": "absolute",
        },
        Object {
          "borderRadius": 0,
          "height": 1,
          "width": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<withMaterialTheme(LabeledCheckbox) label="Label" /> 1`] = `
<View
  accessible={true}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  rippleLocation="tapLocation"
  style={
    Array [
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
        },
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 12,
        },
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          Array [
            undefined,
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginHorizontal": 12,
        },
        Object {
          "marginLeft": 0,
        },
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
        },
      ]
    }
    suppressHighlighting={true}
  >
    Label
  </Text>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "overflow": "hidden",
          "position": "absolute",
        },
        Object {
          "borderRadius": 0,
          "height": 1,
          "width": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<withMaterialTheme(LabeledCheckbox) label="Label" labelColor="#123456" /> 1`] = `
<View
  accessible={true}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  rippleLocation="tapLocation"
  style={
    Array [
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
        },
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 12,
        },
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          Array [
            undefined,
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginHorizontal": 12,
        },
        Object {
          "marginLeft": 0,
        },
        Object {
          "color": "#123456",
        },
      ]
    }
    suppressHighlighting={true}
  >
    Label
  </Text>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "overflow": "hidden",
          "position": "absolute",
        },
        Object {
          "borderRadius": 0,
          "height": 1,
          "width": 1,
        },
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</View>
`;
